scripts/globals/summon.lua

```diff
@@ -10,112 +10,278 @@ function getSummoningSkillOverCap(avatar)
+                local correction = levelDiff * 0.05;
+                local cappedCorrection = math.min(correction, 1.9)
+                cRatio = cRatio + cappedCorrection
+            end
+        end
 
-    -- apply ftp bonus
-    if tpeffect == TP_DMG_BONUS then
-        finaldmg = finaldmg * avatarFTP(skill:getTP(), mtp100, mtp200, mtp300)
-    end
+        --Everything past this point is randomly computed per hit
 
-    returninfo.dmg = finaldmg
-    returninfo.hitslanded = hitslanded
+        numHitsProcessed = 0
 
-    return returninfo
-end
+        local targetHP = target:getHP()
 
--- minFstr = dSTR/4 + 0.5
--- maxFstr = dSTR/4 + 0.25
-function avatarFSTR(att_str, def_vit)
-    local dSTR = att_str - def_vit
-    return math.floor(dSTR / 4 + 0.5), math.floor(dSTR / 4 + 0.25)
-end
+        if firstHitLanded then
+            local wRatio = cRatio
+            local isCrit = math.random() < critRate
+            if isCrit then
+                wRatio = wRatio + 1
+            end
+            -- get a random ratio from min and max
+            local qRatio = getRandRatio(wRatio)
+
+            --Final pDif is qRatio randomized with a 1-1.05 multiplier
+            local pDif = qRatio * (1 + (math.random() * 0.05))
+            --Avatars have Crit Atk Bonus II for +8% crit dmg
+            if isCrit then
+                pDif = pDif * 1.08
+            end
+            finaldmg = avatarHitDmg(weaponDmg, fSTR, pDif) * dmgmod
+            numHitsProcessed = 1
+        end
+
+        -- If the first hit kills the mob other hits are not computed in retail
+        -- This can easily be seen by using BPs on low level mobs where the first hit kills them
+        while numHitsProcessed < numHitsLanded and finaldmg < targetHP do
+            local wRatio = cRatio
+            local isCrit = math.random() < critRate
+            if isCrit then
+                wRatio = wRatio + 1
+            end
+            
+            -- get a random ratio from min and max
+            local qRatio = getRandRatio(wRatio)
+
+            --Final pDif is qRatio randomized with a 1-1.05 multiplier
+            local pDif = qRatio * (1 + (math.random() * 0.05))
+            --Avatars have Crit Atk Bonus II for +8% crit dmg
+            if isCrit then
+                pDif = pDif * 1.08
+            end
+            finaldmg = finaldmg + (avatarHitDmg(weaponDmg, fSTR, pDif) * dmgmodsubsequent)
+            numHitsProcessed = numHitsProcessed + 1
+        end
+
+        -- apply ftp bonus
+        if tpeffect == TP_DMG_BONUS then
+            finaldmg = finaldmg * avatarFTP(skill:getTP(), mtp100, mtp200, mtp300)
+        end
 
-function avatarHitDmg(dmg, pdifMin, pdifMax, fstrMin, fstrMax, critrate)
-    local fstr = math.random(fstrMin, fstrMax)
-    local pdif = math.random(pdifMin * 1000, pdifMax * 1000) / 1000
-    if math.random() < critrate then
-        pdif = math.min(pdif + 1, 4.2)
+        if finaldmg > 0 then
+            target:wakeUp()
+        end
```

Comment:
I know this was here before, but waking up targets is now baked into lua_baseentity:takeDamage(); you can remove it from the raw damage calculation step.